service/scheduler/quartz/src/main/java/org/eclipse/kapua/service/scheduler/trigger/quartz/TriggerServiceImpl.java

```diff
@@ -101,6 +101,10 @@ public Trigger create(TriggerCreator triggerCreator) throws KapuaException {
             throw new RetryIntervalAndCronSelectedException(triggerCreator.getScopeId());
         }
 
+        if(triggerCreator.getStartsOn().getTime() > triggerCreator.getEndsOn().getTime()) {
```

Comment:
Please add 

` ArgumentValidator.notNull(triggerCreator.getStartsOn(), "triggerCreator.startsOn");`

and in this `if` check before for `getEndsOn() != null` otherwise NPE will be thrown. 

Moreover please use `.before()` or `.after()` to compare dates, since it is a native function from `Date` class.

Reply:
@ct-anaalbic this still throws a `NullPointerException` if you create a Trigger with only the `startDate` set.

![screen shot 2019-01-16 at 10 39 02 am](https://user-images.githubusercontent.com/3942036/51239951-fa722580-197a-11e9-9487-ded485b96fe3.png)


```
kapua-console_1  | java.lang.NullPointerException: null
kapua-console_1  | 	at org.eclipse.kapua.service.scheduler.trigger.quartz.TriggerServiceImpl.create(TriggerServiceImpl.java:111)
kapua-console_1  | 	at org.eclipse.kapua.service.scheduler.trigger.quartz.TriggerServiceImpl.create(TriggerServiceImpl.java:62)
kapua-console_1  | 	at org.eclipse.kapua.app.console.module.job.server.GwtTriggerServiceImpl.create(GwtTriggerServiceImpl.java:100)
kapua-console_1  | 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
kapua-console_1  | 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
kapua-console_1  | 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
kapua-console_1  | 	at java.lang.reflect.Method.invoke(Method.java:498)
kapua-console_1  | 	at com.google.gwt.user.server.rpc.RPC.invokeAndEncodeResponse(RPC.java:569)
kapua-console_1  | 	at com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:208)
kapua-console_1  | 	at com.google.gwt.user.server.rpc.RemoteServiceServlet.processPost(RemoteServiceServlet.java:248)
kapua-console_1  | 	at com.google.gwt.user.server.rpc.AbstractRemoteServiceServlet.doPost(AbstractRemoteServiceServlet.java:62)
kapua-console_1  | 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
kapua-console_1  | 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
kapua-console_1  | 	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:865)
kapua-console_1  | 	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1655)
kapua-console_1  | 	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:215)
kapua-console_1  | 	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
kapua-console_1  | 	at org.apache.shiro.web.servlet.AbstractShiroFilter.executeChain(AbstractShiroFilter.java:449)
kapua-console_1  | 	at org.eclipse.kapua.app.console.core.filter.KapuaWebFilter.executeChain(KapuaWebFilter.java:43)
kapua-console_1  | 	at org.apache.shiro.web.servlet.AbstractShiroFilter$1.call(AbstractShiroFilter.java:365)
kapua-console_1  | 	at org.apache.shiro.subject.support.SubjectCallable.doCall(SubjectCallable.java:90)
kapua-console_1  | 	at org.apache.shiro.subject.support.SubjectCallable.call(SubjectCallable.java:83)
kapua-console_1  | 	at org.apache.shiro.subject.support.DelegatingSubject.execute(DelegatingSubject.java:383)
kapua-console_1  | 	at org.apache.shiro.web.servlet.AbstractShiroFilter.doFilterInternal(AbstractShiroFilter.java:362)
kapua-console_1  | 	at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125)
kapua-console_1  | 	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1634)
kapua-console_1  | 	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)
kapua-console_1  | 	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)
kapua-console_1  | 	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)
kapua-console_1  | 	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
kapua-console_1  | 	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)
kapua-console_1  | 	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
kapua-console_1  | 	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
```